src/puppetlabs/puppetdb/cli/benchmark.clj

```diff
@@ -385,175 +376,186 @@
   (jitter (time/minus (time/now) run-interval)
           (time/in-seconds run-interval)))
 
-(defn partition-into-buckets [num-buckets s]
-  (loop [s s
-         buckets (into clojure.lang.PersistentQueue/EMPTY
-                       (repeat num-buckets '()))]
-    (if (seq s)
-      (recur (rest s)
-             (conj (pop buckets)
-                   (cons (first s) (first buckets))))
-      (seq buckets))))
-
-(defn make-jms-nippy-message
-  "Create a jms message containing the clojure data structure 'payload' encoded
-  into a byte array using nippy."
-  [session payload]
-  (let [msg (.createBytesMessage session)
-        payload-bytes (nippy/freeze payload)]
-    (.writeBytes msg payload-bytes)
-    msg))
-
-(defn unpack-jms-nippy-message
-  "Return the clojure data payload from a message created with
-  'make-jms-nippy-message'."
-  [msg]
-  (let [bytes (byte-array (.getBodyLength msg))]
-    (.readBytes msg bytes)
-    (nippy/thaw bytes)))
-
-(deftype AMQBrokerBuffer [^BrokerService broker
-                          ^Session session
-                          ^MessageProducer producer
-                          ^MessageConsumer consumer
-                          mq-dir
-                          queue-count]
+(defn delete-dir-or-report [dir]
+  (try
+    (fs/delete-dir dir)
+    (catch Exception ex
+      (println-err ex))))
+
+(deftype TempFileBuffer [storage-dir q]
   Buffer
   (full? [this] false)
   (remove! [this]
-    (swap! queue-count dec)
-    (mq/commit-or-rollback session (unpack-jms-nippy-message (.receive consumer))))
+    (let [path (.poll q)
+          result (nippy/thaw (Files/readAllBytes path))]
+      (Files/delete path)
+      result))
+
   (add!* [this item]
-    (swap! queue-count inc)
-    (mq/commit-or-rollback session (.send producer (make-jms-nippy-message session item))))
+    (let [path (Files/createTempFile storage-dir "bench-tmp-" ""
+                                     (into-array FileAttribute []))]
+      (Files/write path (nippy/freeze item) (into-array OpenOption []))
+      (.add q path)))
+
   (close-buf! [this]
-    (println "Shutting down AMQ...")
-    (reset! queue-count 0)
-    (amq/stop-broker! broker)
-    (fs/delete-dir mq-dir))
+    (.clear q)
+    (println-err "Cleaning up temp files from" (.toString storage-dir))
+    ;; todo: timeout here
```

Comment:
@rbrw thoughts about what this timeout should be, or how it should be implemented?